packages/push/src/instruction/int/abs.rs

```diff
@@ -0,0 +1,150 @@
+/// cases.
+///
+///    - The "`i64` stack" column indicates the value of the top of the `i64`
+///      stack, or whether it exists.
+///    - The "Success" column indicates whether the instruction succeeds, and if
+///      not what kind of error is returned:
+///       - ✅: success
+///       - ❗: recoverable error, with links to the error kind
+///       - ‼️: fatal error, with links to the error kind
+///    - The "Note" column briefly summarizes the action state in that case
+///
+/// | `i64` stack  |  Success | Note |
+/// | ------------- | ------------- | ------------- |
+/// | `i64::MIN`    | ✅ | `i64::MIN` is replaced with `i64::MAX` |
+/// | exists, not `i64::MIN` | ✅ | Takes the absolute value of the top value of the `i64` stack |
+/// | missing | [❗..](crate::push_vm::stack::StackError::Underflow) | State is unchanged |
+///
+/// # Errors
+///
+/// Returns a
+/// [`StackError::Underflow`](crate::push_vm::stack::StackError::Underflow)
+/// error when the `i64` stack is empty.
+///
+/// # Differences
+/// Implementations of integer negation instructions in Clojure (e.g., Clojush
+/// or Propeller) or Python (e.g. `PyshGP`) won't have the wrapping issue
+/// because they act on arbitrary precision integers.
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq)]
+pub struct Abs;
+
+impl<S> Instruction<S> for Abs
+where
+    S: Clone + HasStack<i64>,
+{
+    type Error = PushInstructionError;
+
+    fn perform(&self, state: S) -> InstructionResult<S, Self::Error> {
+        let int_stack = state.stack::<i64>();
+        int_stack
+            .top()
+            .map(|x| x.saturating_abs())
+            .replace_on(1, state)
+    }
+}
+
+#[cfg(test)]
+#[rustversion::attr(before(1.81), allow(clippy::unwrap_used))]
+#[rustversion::attr(
+    since(1.81),
+    expect(
+        clippy::unwrap_used,
+        reason = "Panicking is the best way to deal with errors in unit tests"
+    )
+)]
+mod tests {
+    use proptest::prop_assert_eq;
+    use test_strategy::proptest;
+
+    use crate::{
+        instruction::{Instruction, IntInstruction},
+        push_vm::{push_state::PushState, HasStack},
+    };
+
+    // We need to make sure `Abs` properly handles the
+    // case where the value we're the absolute value of is `i64::MIN`.
+    // Simply taking the absolute value of `i64::MIN` will generate an overflow
+    // error because `i64::MIN` has a larger magnitude than
+    // the largest representable positive value (`i64::MAX`).
+    // We want to using a saturating version of absolute value
+    // that converts `i64::MIN` to `i64::MAX`.
+    #[test]
+    fn abs_with_i64_min() {
+        let input = i64::MIN;
+        let state = PushState::builder()
+            .with_max_stack_size(1)
+            .with_int_values(std::iter::once(input))
+            .unwrap()
+            .with_no_program()
+            .build();
+        let result = IntInstruction::abs().perform(state).unwrap();
```

Comment:
Please make sure to not rely on the "instruction group" for tests for a particular instruction:
```suggestion
        let result = Abs.perform(state).unwrap();
```

Reply:
Good catch – thanks. We might want to fix that on `Negate` since we're already changing things there.